Sources/MongoSwift/MongoCredential.swift

```diff
@@ -0,0 +1,71 @@
+/// Represents an authentication credential.
+public struct MongoCredential: Decodable, Equatable {
+    /// A string containing the username. For auth mechanisms that do not utilize a password, this may be the entire
+    /// `userinfo` token from the connection string.
+    public var username: String?
+
+    /// A string containing the password.
+    public var password: String?
+
+    /// A string containing the authentication database.
+    public var source: String?
+
+    /// The authentication mechanism. A nil value for this property indicates that a mechanism wasn't specified and
+    /// that mechanism negotiation is required.
+    public var mechanism: Mechanism?
+
+    /// A document containing mechanism-specific properties.
+    public var mechanismProperties: Document?
+
+    private enum CodingKeys: String, CodingKey {
+        case username, password, source, mechanism, mechanismProperties = "mechanism_properties"
+    }
+
+    public init(
+        username: String? = nil,
+        password: String? = nil,
+        source: String? = nil,
+        mechanism: Mechanism? = nil,
+        mechanismProperties: Document? = nil
+    ) {
+        self.username = username
+        self.password = password
+        self.source = source
+        self.mechanism = mechanism
+        self.mechanismProperties = mechanismProperties
+    }
+
+    /// Possible authentication mechanisms.
+    public struct Mechanism: Codable, Equatable {
+        /// See https://docs.mongodb.com/manual/core/kerberos/
+        public static let gssAPI = Mechanism(name: "GSSAPI")
+        /// Deprecated: see https://docs.mongodb.com/manual/release-notes/3.0-scram/
```

Comment:
should we remove MongoDB-CR support altogether? it was deprecated in 3.0 and removed from the server altogether in 4.0. cc @mbroadst

Reply:
Yes, let's go ahead and remove it. We can add it back if someone asks for it, but I can't imagine who would